src/phasorpy/phasor.py

```diff
@@ -0,0 +1,507 @@
+"""Manipulate phasor  data
+
+The ``phasorpy.phasor`` module provides functions to:
+    - transform phasor to polar coordinates
+    - calibration of phasor coordinates
+    - compute phasor coordinates from lifetime values
+
+
+Lifetime values for known fluorophores can be obtained from the ISS website:
+https://iss.com/resources#lifetime-data-of-selected-fluorophores.
+
+"""
+
+from __future__ import annotations
+
+__all__ = [
+    "phasor_calibration",
+    "polar_from_reference_phasor",
+    "polar_from_reference",
+    "phasor_to_polar",
+    "phasor_from_lifetime",
+    "phasor_center",
+    "_mean",
+    "_spatial_median",
+    "_geometric_median",
+]
+
+from typing import TYPE_CHECKING
+
+if TYPE_CHECKING:
+    from ._typing import Any, NDArray, ArrayLike, Literal
+
+import math
+
+import numpy
+
+
+def phasor_calibration(
+    real: ArrayLike,
+    imag: ArrayLike,
+    phase0: ArrayLike = 0.0,
+    modulation0: ArrayLike = 1.0,
+    /,
+) -> tuple[ArrayLike, ArrayLike]:
+    """Return calibrated/referenced phasor coordinates.
+
+    Parameters
+    ----------
+    real : array_like
+        Scalar or array containing the real components of phasor coordinates.
+    imag : array_like
+        Scalar or array containing the imaginary components of phasor
+        coordinates.
+    phase0 : array_like, optional
+        Angular component of polar coordinates for calibration. Defaults
+        to 0.0.
+    modulation0 : array_like, optional
+        Radial component of polar coordinates for calibration. Defaults to 1.0.
+
+
+    Returns
+    -------
+    tuple[NDArray[Any], NDArray[Any]]
+        real:
+            Calibrated real component of phasor coordinates in Cartesian form.
+        imag:
+            Calibrated imaginary component of phasor coordinates in Cartesian
+            form.
+
+    Examples
+    --------
+    >>> real_data = numpy.array([1.0, 2.0, 3.0])
+    >>> imag_data = numpy.array([4.0, 5.0, 6.0])
+    >>> phase0 = 0.5
+    >>> modulation0 = 2.0
+    >>> calibrated_real, calibrated_imag = phasor_calibration(
+    ...     real_data, imag_data,
+    ...     phase0,
+    ...     modulation0
+    ... )
+    >>> calibrated_real
+    array(...)
+    >>> calibrated_imag
+    array(...)
+    """
+    phi0, mod0 = numpy.asarray(phase0), numpy.asarray(modulation0)
+    if numpy.all(phi0 == 0) and numpy.all(mod0 == 1):
+        return real, imag
+    real = numpy.asarray(real)
+    imag = numpy.asarray(imag)
+    correction_matrix = (
+        numpy.array(
+            (
+                (numpy.cos(phi0), -numpy.sin(phi0)),
+                (numpy.sin(phi0), numpy.cos(phi0)),
+            )
+        )
+        * mod0
+    )
+    real, imag = (
+        correction_matrix.dot(numpy.vstack([real.flatten(), imag.flatten()]))
+    ).reshape((2, *real.shape))
```

Comment:
In the most general case one would want to be able to calibrate every item/pixel of the polar coordinate arrays separately with different reference coordinates. The instrumentation might require that or the phasor coordinates might have a frequency axis. Also, this library is not only intended to process measured images, but also to generate/process synthetic data.

Reply:
Looking at this implementation, the other functions, and their default use cases, we might want to consider passing/returning coordinate components as shaped `(2, ...)` arrays instead of separate arrays. Internally, that should be simpler and more efficient, but what about users interacting with the library? Thoughts?

This function signature would would simplify to:
```Python
def phasor_calibration(
    phasor: ArrayLike,
    polar0: ArrayLike = (0.0, 1.0)
    /,
) -> NDArray[Any]:
    """Return calibrated/referenced phasor coordinates.

    Parameters
    ----------
    phasor : array_like
        Phasor coordinates to be corrected.
    polar0 : array_like, optional
        Phase and modulation for calibration.

    Returns
    -------
    phasor: ndarray
        Calibrated phasor coordinates.

    Examples
    --------
    >>> phasor = numpy.array([[1.0, 2.0, 3.0], [4.0, 5.0, 6.0]])
    >>> phasor_calibration(phasor, (0.5, 2.0))
    array(...)
    """
    phi0, mod0 = _coordinates(polar0)
    phasor = _coordinates(phasor)
    ...
```

Reply:
I really like that idea. How will the `coordinates` work? Is it like a private helper function that unravels the different components?

I am still trying to make it work with phi0 and mod0 as arrays, so I will update later if I find the solution.

Reply:
Hold on, my comment was intended to start a discussion, not to change the implementation at this point. There are pro and cons for the approaches...

Reply:
If it is too complicated to implement some cases using numpy array computing, raise a NotImplementedError or ValueError in those cases for now. Similar to what is in `phasor_from_lifetime`. I think sooner than later we will have to start using Cython to implement some of the low level conversions for efficiency and generalization. But it should probably not delay this PR.